src/js/components/hooks/useDelayedMount.js

```diff
@@ -1,17 +1,19 @@
 /* @flow */
-import {useEffect, useRef, useState} from "react"
+import {useEffect, useState} from "react"
 
 export default function useDelayedMount(show: boolean, delay: number) {
   let [ready, setReady] = useState(false)
-  let id = useRef()
 
   useEffect(() => {
+    let id
     if (show) {
-      id.current = setTimeout(() => setReady(true), delay)
+      id = setTimeout(() => setReady(true), delay)
     } else {
-      clearTimeout(id.current)
       setReady(false)
     }
+    return () => {
+      clearTimeout(id)
+    }
```

Comment:
This fixes that warning we'd see in the console at times from the SearchSpinner.